comms/ModFileRepository.cpp

```diff
@@ -120,10 +120,15 @@ class CModFileRepository : public IRepository, public clib::CLockableUnknown
     {
         IModuleVector allModules;
         GetAllModules(allModules, GetChecksum, noRefresh);
+        typedef std::vector<std::_tstring> split_vector_type;
         for(IModuleVector::iterator itr = allModules.begin(); itr != allModules.end(); ++itr)
         {
             IModule * parent = itr->get()->GetParentModule();
-            if ((!parent && module.empty()) || (parent && boost::algorithm::equals(module, parent->GetQualifiedLabel())))
+            std::_tstring label = itr->get()->GetQualifiedLabel();
+            split_vector_type lableParts;
+            boost::algorithm::split(lableParts, label, boost::algorithm::is_any_of(CModuleHelper::DELIMTER), boost::algorithm::token_compress_on);
+
+            if ((parent && lableParts.size() > 1) || (!parent && module.empty()) || (parent && boost::algorithm::equals(module, parent->GetQualifiedLabel())))
```

Comment:
This only checks up 1 level - don't we need to walk the entire "labelParts" and each level to the module list?

Reply:
@GordonSmith I will check this more thoroughly.